packages/webapp/src/pages/delegates/index.tsx

```diff
@@ -22,80 +33,128 @@ interface Props {
 const MEMBERS_PAGE_SIZE = 18;
 
 export const DelegatesPage = (props: Props) => {
-    const { isError: isMembersDataFetchError, data: members } = useQuery({
+    const [ualAccount] = useUALAccount();
+    const accountName = ualAccount?.accountName;
+    const { data: currentMember } = useCurrentMember();
+
+    const { data: members } = useQuery({
         ...queryMembers(1, MEMBERS_PAGE_SIZE),
         keepPreviousData: true,
     });
 
-    const {
-        isError: isMemberStatsDataFetchError,
-        data: membersStats,
-    } = useQuery({
+    const loggedInMemberName = currentMember?.name || accountName;
+
+    const { data: membersStats } = useQuery({
         ...queryMembersStats,
         keepPreviousData: true,
     });
 
+    const { data: leadRepresentative } = useQuery({
+        ...queryHeadDelegate,
+        keepPreviousData: true,
+    });
+
+    const { data: myDelegation } = useQuery({
+        ...queryMyDelegation(members!, loggedInMemberName),
+        enabled: !!members && !!loggedInMemberName,
+        keepPreviousData: true,
+    });
+
+    if (
+        !loggedInMemberName ||
+        !members ||
+        !membersStats ||
+        !leadRepresentative ||
+        !myDelegation
+    ) {
+        return <Text size="lg">Fetching Data...</Text>;
+    }
+
+    const loggedInMember: MemberData | undefined =
+        members && getMemberRecordFromName(members, loggedInMemberName);
+    if (!loggedInMember) {
+        return <Text size="lg">Failed to get loggedInMember record</Text>;
```

Comment:
Might be easier to jump around and test, log out and back in with different users, etc. if you wrap this return and the one above it in a `<RawLayout...` too.

Reply:
So I tried this logged in as Egeon and I hit this "Failed to get loggedInMember record" even though he is a member. The reason is that `getMemberRecordFromName` expects a blockchain account name (`egeon.edev`)...not the friendly member name (`Egeon The Great`). `currentMember?.name` is a friendly name.

The only reason it's already working with `Eden Member 11` is that `useCurrentMember` is returning `undefined` because it's actually querying the chain for the associated eden member, but `Eden Member 11` only exists as a fixture, not in the actual chain.

If you get rid of `useCurrentMember` on line `38` and get rid of `loggedInMemberName` on `46` entirely, and instead simply pass in the `accountName` from `37` in lieu of `loggedInMemberName`, it starts working again as expected--for both Eden Member 11 and our friend Egeon.